packages/compartment-mapper/src/node-modules.js

```diff
@@ -364,17 +364,17 @@ const translateGraph = (
   // The full map includes every exported module from every dependencey
   // package and is a complete list of every external module that the
   // corresponding compartment can import.
-  for (const [
-    packageLocation,
-    { label, dependencies, parsers, types },
-  ] of entries(graph)) {
+  for (const packageLocation of keys(graph).sort()) {
+    const { label, dependencies, parsers, types } = graph[packageLocation];
     /** @type {Record<string, ModuleDescriptor>} */
     const modules = {};
     /** @type {Record<string, ScopeDescriptor>} */
     const scopes = {};
-    for (const [dependencyName, packageLocation] of entries(dependencies)) {
+    for (const dependencyName of keys(dependencies).sort()) {
+      const packageLocation = dependencies[dependencyName];
       const { exports, explicit } = graph[packageLocation];
-      for (const [exportName, module] of entries(exports)) {
+      for (const exportName of keys(exports).sort()) {
```

Comment:
it seems tempting to factor out this pattern as `entriesByKey: (obj: Record<string, T>) => [string, T][]`